pkg/shell/script.go

```diff
@@ -24,7 +24,10 @@ type ScriptConfig struct {
 
 // Script executes a shell script.
 func Script(fds [3]*os.File, args []string, cfg *ScriptConfig) int {
-	defer rescue()
+	// For non-interactive shells we want elvish specific output from a panic
+	// but do not want to launch a rescue shell.
+	PanicRescueShell = false
+	defer handlePanic()
```

Comment:
It's simpler to just not do `defer handlePanic` here. Setting a global here is also fragile.

In case you are keeping this because we *might* want to add more handling than the default panic handler in future, [YAGNI](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it).

Reply:
I'm familiar with the YAGNI principal. The first major project I worked on at Google had a couple of "features" that greatly complicated the implementation and were not usable for various reasons. Those features were implemented because the author thought they might be useful. I eliminated one of them almost immediately after joining the project because the feature was inherently unusable and causing problems when people tried to use it. The design of the feature didn't address what the users needed and the implementation guaranteed unpredictable behavior.

My hope in doing it this way was to eventually add additional context to the panic data in a subsequent change. Most notably including an Elvish call stack. But after pondering the issue some more I see that isn't going to be practical for several reasons.

Having said all that this particular global flag is not "fragile". Not every global var, or singleton object, is inherently problematic.

Reply:
It *is* problematic - if a program calls `Script` and then `Interact`, the panic handler is turned off for the second `Interact` call.